db/migrate/20220225074245_create_users.rb

```diff
@@ -0,0 +1,14 @@
+class CreateUsers < ActiveRecord::Migration[6.1]
+  def change
+    create_table :users do |t|
+      t.string :first_name
+      t.string :last_name
+      t.string :username
+      t.string :password_hash
+      t.string :phone_number
+      t.string :email
```

Comment:
I would expect we want `username`, `password_hash` and `email` to always have a value, so we should set up our schema base on these expectations:
```ruby
# non-nullable fields should always have a corresponding default, 
# especially if they're being added to an existing model
t.string :username, null: false, default: ""
t.string :password_hash, null: false, default: "" # there may be a case for null passwords, but unlikely here
t.string :email, null: false, default: ""
```
I know it's split over two commits for the purpose of the tutorial, but normally we'd include our indexes and basic validations as part of the model creation, unless there's a good reason not to.

Reply:
That would definitely make sense, I was unsure as to the approach there, e.g., if we simply let rails handle it via object creation validators ensuring a value is present upon creation or we handle it both on the schema-end and rails-end. This has definitely answered that, thank you!

In regards to the split commits, I definitely get your approach there - however after our last discussion I tried to keep commits simplified to their relevant section for the tutorial, but I'll aim to fix that up too!